### ISSUE TITLE
Add page verification checks for axe tests

### ISSUE THREAD

**Issue by TobyCyan** [TEAMMATES/teammates#13927]
Axe tests do not verify that the currently opened page is the intended one before running accessibility checks.

As a result, the tests may report accessibility violations on an incorrect page not intended for real users (i.e. devServerLoginPage), which can make the failures misleading and difficult to debug.
